packages/zwave-js/src/lib/commandclass/PowerlevelCC.ts

```diff
@@ -0,0 +1,409 @@
+		});
+		const response = await this.driver.sendCommand<PowerlevelCCTestNodeReport>(
+			cc,
+			this.commandOptions,
+		);
+		if (response) {
+			return pick(response, [
+				"testNodeId",
+				"testStatus",
+				"testFrameAcknowlegedCount",
+			]);
+		}
+	}
+}
+
+@commandClass(CommandClasses.Powerlevel)
+@implementedVersion(1)
+export class PowerlevelCC extends CommandClass {
+	declare ccCommand: PowerlevelCommand;
+}
+
+type PowerlevelCCSetOptions = CCCommandOptions &
+	(
+		| {
+				powerlevel: Powerlevel;
+				timeout: number;
+		  }
+		| {
+				powerlevel: typeof Powerlevel.NormalPower;
+		  }
+	);
+
+@CCCommand(PowerlevelCommand.Set)
+export class PowerlevelCCSet extends PowerlevelCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | PowerlevelCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			if ("timeout" in options) {
+				if (options.timeout < 1 || options.timeout > 255) {
+					throw new ZWaveError(
+						`${this.constructor.name}: The timeout parameter must be between 1 and 255.`,
+						ZWaveErrorCodes.Argument_Invalid,
+					);
+				}
+				this.timeout = options.timeout;
+			}
+			this.powerlevel = options.powerlevel;
+		}
+	}
+
+	public powerlevel: Powerlevel;
+	public timeout?: number;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([this.powerlevel, this.timeout ?? 0x00]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		const message: MessageRecord = {
+			powerlevel: getEnumMemberName(Powerlevel, this.powerlevel),
+		};
+		if (this.timeout != undefined) {
+			message.timeout = this.timeout;
+		}
+		return {
+			...super.toLogEntry(),
+			message,
+		};
+	}
+}
+
+@CCCommand(PowerlevelCommand.Report)
+export class PowerlevelCCReport extends PowerlevelCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		this.powerlevel = this.payload[0];
+		if (this.powerlevel !== Powerlevel.NormalPower) {
+			this.timeout = this.payload[1];
+		}
+
+		this.persistValues();
+	}
+
+	@ccValue({ stateful: false })
+	@ccValueMetadata({
+		...ValueMetadata.UInt8,
+		states: enumValuesToMetadataStates(Powerlevel),
+		label: "Power level",
+		description: "The current power level  in effect on the node.",
+	})
+	public readonly powerlevel: Powerlevel;
+
+	@ccValue({ stateful: false })
+	@ccValueMetadata({
+		...ValueMetadata.UInt8,
+		min: 1,
+		max: 255,
+		label: "Timeout",
+		description: "Time in seconds before node returns to Normal power.",
+	})
+	public readonly timeout?: number;
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: { "Power level": this.powerlevel, Timeout: this.timeout },
```

Comment:
I think all other of these messages begin lowercase - lets keep it this way.
Also, timeout should only be logged if not undefined.